tests/Backend/AMQPBackendTest.php

```diff
@@ -107,44 +107,38 @@ public function testInitializeWithDeadLetterExchangeAndNoDeadLetterRoutingKey():
         $deadLetterTopic = new ImplAmqpTopic(self::DEAD_LETTER_EXCHANGE);
 
         $contextMock = $this->createMock(AmqpContext::class);
-        $contextMock->expects($this->at(0))
+        $contextMock->expects($this->once())
             ->method('createQueue')
             ->with($this->identicalTo(self::QUEUE))
             ->willReturn($queue);
 
-        $contextMock->expects($this->at(1))
+        $contextMock->expects($this->once())
             ->method('declareQueue')
             ->with($this->identicalTo($queue))
             ->willReturnCallback(function (AmqpQueue $queue) {
                 $this->assertTrue((bool) ($queue->getFlags() & AmqpQueue::FLAG_DURABLE));
                 $this->assertSame(['x-dead-letter-exchange' => self::DEAD_LETTER_EXCHANGE], $queue->getArguments());
             });
 
-        $contextMock->expects($this->at(2))
+        $contextMock->expects($this->exactly(2))
             ->method('createTopic')
-            ->with($this->identicalTo(self::EXCHANGE))
-            ->willReturn($topic);
-
-        $contextMock->expects($this->at(3))
-            ->method('declareTopic')
-            ->with($this->identicalTo($topic));
+            ->willReturnMap([
+                [self::EXCHANGE, $topic],
+                [self::DEAD_LETTER_EXCHANGE, $deadLetterTopic],
+            ]);
 
-        $contextMock->expects($this->at(4))
+        $contextMock->expects($this->atLeastOnce())
             ->method('bind')
             ->with($this->isInstanceOf(AmqpBind::class));
 
-        $contextMock->expects($this->at(5))
-            ->method('createTopic')
-            ->with($this->identicalTo(self::DEAD_LETTER_EXCHANGE))
-            ->willReturn($deadLetterTopic);
-
-        $contextMock->expects($this->at(6))
+        $contextMock->expects($this->exactly(2))
```

Comment:
Was the previous expected call count wrong?

Reply:
Yes, I didnt know the at() works like that, but it is deprecated because you shouldnt rely on the order